### PR TITLE
Limit inline values to current function-scope

### LINKED ISSUE #8: Feature request: Only show inline values inside the current scope
There should be an option (or default behavior?) to only show inline values for the current scope. The view can get filled with unnecessary noise and cause confusion for similar parameters/variables in other places of the file.

The extra processing of parsing the whole file might also affect performance when working with files with hundreds or thousands of lines like in a combined .psm1 module-file.

Example of current behavior. Showing the variable outside t3 might be confusing as it would never be applicable there
inline-scopes

### PR DESCRIPTION
**Pull request by fflaten**
The extension only shows values for variables in local scope. To reduce noise and wrong values (local values shown for matching variables in parent scopes) this PR filters the lines that are being scanned to only cover relevant parts.

Use function start as startline when stopped inside a function, else document start as before. Always exclude other functions.

- [x] Setting to control behavior
  - [ ] Remove? Unnecessary?
- [x] Calculate startLine (document or function start) depending on stopped location
- [x] Always exclude lines for functions not in scope
- [x] Add tests
- [x] Test for performance impact and optimize. Especially for large files (tested against `Pester.psm1` ~17k lines)

Fix #8

### CODE REVIEW

**Comment by fflaten** [comment]
@TylerLeonhardt Any way to profile this type of extension? `Start Extension Host Profile` doesn't show the methods in this code. Because it's a provider?

**Comment by fflaten** [comment]
I believe this is ready for review.
- Configuration option. Text, name etc. ok?
  - Should it even be a setting considering how many wrong answers that can be shown in parent scopes?
- General naming and structure. Both for code and tests.
- Optimizations?
- Make get* functions public + unit tests?
- How to apply MIT license for borrowed code in `testUtils.ts` ?

**Comment by TylerLeonhardt** [comment]
I noticed you had a couple other commits. Ready for me now @fflaten?

**Comment by fflaten** [comment]
Yes please. Ready for review, but not merge as I'm missing a MIT license if not more depending on comments.

Glad I did the last commits while I waited though as it went from 11 to 1 sec test time 😄

File: src/documentParser.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return res;
+    }
+
+    async getFunctionsInDocument(document: vscode.TextDocument): Promise<vscode.DocumentSymbol[]> {
+        const cacheKey = document.uri.toString();
+        if (this.functionCache.has(cacheKey)) {
+            return this.functionCache.get(cacheKey)!;
+        }
+
+        const documentSymbols = await vscode.commands.executeCommand<vscode.DocumentSymbol[]>('vscode.executeDocumentSymbolProvider', document.uri);
```
Comment (TylerLeonhardt) [inline src/documentParser.ts:36]: clever to use the document symbol provider

**Review by TylerLeonhardt — APPROVED** [review]

**Comment by TylerLeonhardt** [comment]
Awesome work! Published. It should take a few minutes to show up

**Comment by fflaten** [comment]
Thanks 👏
Did you have any thoughts on the questions [above](https://github.com/TylerLeonhardt/vscode-inline-values-powershell/pull/14#issuecomment-1155623091)? Like if the setting is neccessary at all?

If https://github.com/PowerShell/PowerShellEditorServices/pull/1886 gets merged it would be natural to update this to also track/scope to class, configuration or method/constructor in addition to functions.

**Comment by TylerLeonhardt** [comment]
I think leaving the setting there for now is totally ok. Let's see if anyone complains about the new behavior and if they do, they have an easy way to revert back. If no one complains, then we can take it out :)

**Pull request merged**